position_in_partition.hh

```diff
@@ -572,6 +598,25 @@ public:
     }
 };
 
+struct view_and_holder {
+    position_in_partition_view view;
+    std::optional<position_in_partition> holder;
+};
+
+inline
+view_and_holder position_in_partition_view::after_key(const schema& s, position_in_partition_view pos) {
+    if (!pos.is_clustering_row()) {
+        return {pos, std::nullopt};
+    }
+    if (pos.key().is_full(s)) {
+        return {position_in_partition_view::after_all_prefixed(pos.key()), std::nullopt};
+    }
+    // FIXME: This wouldn't be needed if we had a bound weight to represent this.
```

Comment:
I think we should do this, but not in this PR, as we will likely want to backport this fix.
We can easily locate all sites which look at weight explicitely and consider the effect of the new weights for each one. There shouldn't be a lot of them, most should be in this very file.